### ISSUE TITLE
Explore Luxon and do a POC of how we can integrate it in the library

### ISSUE THREAD

**Issue by InnaAtanasova** [SAP/fundamental-ngx#3373]
Link to Luxon https://moment.github.io/luxon/ 

- Align with Valor team 
- `Adapter` in documentation

**Comment by mikerodonnell89** [comment]
Here are my takeaways from what I've learned

One main advantage I think would be Luxon's Duration class, I think this could be useful when working with date ranges.  Check it out here: https://moment.github.io/luxon/docs/class/src/duration.js~Duration.html

Some other significant differences that I would not consider good nor bad would be Luxon's immutability.  Moment is mutable whereas Luxon is not.  So in moment, you'd get this:

```
var m1 = moment();
var m2 = m1.add(1, 'hours');
m1.valueOf() === m2.valueOf(); //=> true
```

And in Luxon, you'd get
```
var d1 = DateTime.local();
var d2 = d1.plus({ hours: 1 });
d1.valueOf() === d2.valueOf(); //=> false
```

Not a big deal, and if you consider the latter advantageous then you could accomplish immutability using Moment and an extra line of code.

In luxon, months are 1 indexed.  Possibly easier to read, but Moment's 0-indexed months aren't that hard to understand for anyone who's worked with arrays (everyone)

Luxon uses getters instead of access methods.  So in Moment you say `.year()` to get the year but it's just `.year` in luxon.  Lots of identical functions get renamed (see the above example with add/plus).  I think these minor differences are just more of a hassle than anything.

In Luxon, localization is implemented by the native Intl API.  That has the potential to be advantageous because the developer approaching Luxon could possibly already be familiar with this api.  But I think that's unlikely - in my anecdotal experience, many web devs are already familiar with moment's locale api.

Long story short, I think that because Moment is so ubiquitous with handling dates and times in web development, and that Luxon does not provide significant improvements over Moment, we should just stick with Moment
